### PR TITLE
[home-assistant] don't restrict to rsa ssh key type

### PR DESCRIPTION
**Pull request by Diaoul**
**Description of the change**
Don't restrict to rsa ssh key type for home-assistant

**Benefits**
Not much.

**Possible drawbacks**
Not much either.

**Applicable issues**
I prefer ed25519.

**Additional information**
I could've used RSA for this TBH but I decided to make a PR since I have a big heart.

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [x] (optional) Chart is using our [common](https://github.com/k8s-at-home/charts/tree/master/charts/common) library as a dependency.
- [x] (optional) Variables are documented in the README.md

### CODE REVIEW

File: charts/home-assistant/templates/secret.yaml [hunk 1]
```diff
@@ -8,8 +8,8 @@ metadata:
 type: Opaque
 data:
   {{- if .Values.git.deployKey }}
-  id_rsa: {{ .Values.git.deployKey | b64enc | quote }}
+  id_{{ .Values.git.keyType }}: {{ .Values.git.deployKey | b64enc | quote }}
   {{- else }}
-  id_rsa: {{ .Values.git.deployKeyBase64 | quote }}
+  id_{{ .Values.git.keyType }}: {{ .Values.git.deployKeyBase64 | quote }}
```
Comment (Diaoul) [inline charts/home-assistant/templates/secret.yaml:13]: Please double check my space management here.

File: charts/home-assistant/README.md [hunk 2]
```diff
@@ -229,4 +230,4 @@ This is the last version before starting this changelog. All sorts of cool stuff
 - Join our [Discord](https://discord.gg/sTMX7Vh) community
 
 ----------------------------------------------
-Autogenerated from chart metadata using [helm-docs v1.5.0](https://github.com/norwoodj/helm-docs/releases/v1.5.0)
\ No newline at end of file
+Autogenerated from chart metadata using [helm-docs v1.5.0](https://github.com/norwoodj/helm-docs/releases/v1.5.0)
```
Comment (Diaoul) [inline charts/home-assistant/README.md:233]: :shrug:

**Review by onedr0p — APPROVED** [review]

**Comment by onedr0p** [comment]
I'm actually curious if this is needed, won't you be able to update the mount path in the code addon to point to id_edsa?

https://github.com/onedr0p/home-cluster/blob/cae2e6d704fd0bfee4c56b5174abc86b585aa602/cluster/home/home-assistant/helm-release.yaml#L96

**Comment by Diaoul** [comment]
Probably yes. This secret name just feels weird when my using rsa. Maybe it
would be best to name it something that does not imply the key type and
make a breaking change.

**Comment by onedr0p** [comment]
I am closing this as we discussed it really comes down to the name of the key. I don't know if we want to roll a breaking change to update it tbh.

**Comment by Diaoul** [comment]
Fine by me. Could it be documented somewhere that the key created is named `id_rsa`? I had to dig into the templates to find that information.

**Comment by Diaoul** [comment]
Also just for correctness, it is not a breaking change since the default `keyType` is `rsa` so the key is still named `id_rsa` unless you specify `keyType` to be something else.

**Pull request closed without merging**